Generate typed PocketBase type (#82)
patmood/pocketbase-typegen: README.md

  - `[CollectionName][FieldName]Options` If the collection contains a select field with set values, an enum of the options will be generated.
- `CollectionRecords` A type mapping each collection name to the record type.
- `CollectionResponses` A type mapping each collection name to the response type.
- `TypedPocketBase` A type for usage with type asserted PocketBase instance.

## Example Usage

> In [PocketBase SDK](https://github.com/pocketbase/js-sdk) v0.18.3+ you can use the PocketBase SDK with type assertion to have collections automatically typed:

Reviewer: ```suggestion
Using PocketBase SDK v0.18.3+, collections can be [automatically typed](https://github.com/pocketbase/js-sdk#specify-typescript-definitions) using the generated `TypedPocketBase` type:
```
Slight wording suggestion

Author: Much better :smile:

```typescript
import { TypedPocketBase } from "./pocketbase-types"

const pb = new PocketBase('http://127.0.0.1:8090') as TypedPocketBase

await pb.collection('tasks').getOne("RECORD_ID") // -> results in TaskResponse
await pb.collection('posts').getOne("RECORD_ID") // -> results in PostResponse
```

> In [PocketBase SDK](https://github.com/pocketbase/js-sdk) v0.8+ you can use generic types when fetching records, eg:

Reviewer: ```suggestion
Alternatively, you can use generic types for each request, eg:
```
Nobody new will be using <v0.8 now
